### PR TITLE
chore: Updated Mocha to 10.2.0 

### LINKED ISSUE #4527: [NV-3028] Upgrade mocha to latest versions on all app packages
### What?
Upgrade mocha to latest version `10.2.0` on all apps packages

### Definition of Done
- Tests in all affected packages are passing
- pnpm lock file is updated
- package.json in all relevant packages is updated

<sub>[NV-3028](https://linear.app/novu/issue/NV-3028/upgrade-mocha-to-latest-versions-on-all-app-packages)</sub>

### PR DESCRIPTION
**Pull request by ayushshukla7777**
### What change does this PR introduce?
chore: Update Mocha to the latest version - 10.2.0 and  Updated @types/mocha to the latest version - 10.0.2

### Why was this change needed?
closes #4527 

### Other information (Screenshots)
image

### CODE REVIEW

File: providers/fcm/package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
   "devDependencies": {
     "@istanbuljs/nyc-config-typescript": "^1.0.1",
     "@types/jest": "^29.5.0",
-    "@types/mocha": "^8.0.1",
+"@types/mocha": "^10.0.2",
```
Comment (scopsy) [inline providers/fcm/package.json:41]: Seems like a tab issue here :)

Comment (ayushshukla7777) [reply providers/fcm/package.json]: hey @scopsy ,  Corrected the tab issue

**Comment by scopsy** [comment]
@ayushshukla7777 could you please update from next and re run pnpm install to resolve the conflicts there?

**Comment by ayushshukla7777** [comment]
Hey @scopsy , I have updated and pushed new commit as you said

**Review by scopsy — APPROVED** [review]

**Pull request merged**